File: libs/eave-stdlib-py/src/eave/stdlib/core_api/models/forge.py
import enum

from eave.stdlib.core_api.models import BaseInputModel, BaseResponseModel

import pydantic
from typing import Optional



class EaveForgeInboundOperation(enum.StrEnum):
    createDocument = "createDocument"
    updateDocument = "updateDocument"
    archiveDocument = "archiveDocument"


class ForgeInstallation(BaseResponseModel):
    id: pydantic.UUID4
    forge_app_id: str
    forge_app_version: str
    forge_app_installation_id: str
    forge_app_installer_account_id: str
    webtrigger_url: str
    confluence_space_key: Optional[str]


class QueryForgeInstallationInput(BaseInputModel):
    forge_app_id: str
    forge_app_installation_id: str


class RegisterForgeInstallationInput(BaseInputModel):
    """
    These field names MUST match the field names defined in the ORM.
    It is recommended to not change these.
    """

    forge_app_id: str
    forge_app_version: str
    forge_app_installation_id: str
    forge_app_installer_account_id: str
    webtrigger_url: str
    confluence_space_key: Optional[str]


class UpdateForgeInstallationInput(BaseInputModel):
    """
    These field names MUST match the field names defined in the ORM.
    It is recommended to not change these.
    """

    forge_app_installation_id: str
    forge_app_version: Optional[str]
    forge_app_installer_account_id: Optional[str]
    webtrigger_url: Optional[str]
    confluence_space_key: Optional[str]

File: libs/eave-stdlib-ts/src/core-api/models/forge.ts

export enum EaveForgeInboundOperation {
  createDocument = 'createDocument',
  updateDocument = 'updateDocument',
  archiveDocument = 'archiveDocument',

}

export type ForgeInstallation = {
  id: string;
  forge_app_id: string;
  forge_app_version: string;
  forge_app_installation_id: string;
  forge_app_installer_account_id: string;
  webtrigger_url: string;
  confluence_space_key?: string;
}

export type QueryForgeInstallationInput = {
  forge_app_id: string;
  forge_app_installation_id: string;
}
export type RegisterForgeInstallationInput = {
  forge_app_id: string;
  forge_app_version: string;
  forge_app_installation_id: string;
  forge_app_installer_account_id: string;
  webtrigger_url: string;
  confluence_space_key?: string;
}

export type UpdateForgeInstallationInput = {
  forge_app_installation_id: string;
  forge_app_version?: string;
  forge_app_installer_account_id?: string;
  webtrigger_url?: string;
  confluence_space_key?: string;
}